validation/management/commands/autoval.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python3
+# -*- coding: UTF-8 -*-
+
+# Copyright 2021 Jolene Poulin <jcpoulin@ualberta.ca>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from django.conf import settings  # type: ignore
+from django.core.management.base import BaseCommand
+
+from validation.helpers import (
+    perfect_match,
+    exactly_one_analysis,
+    get_distance_with_translations,
+)
+from validation.models import Phrase
+
+from tqdm import tqdm
+
+
+class Command(BaseCommand):
+    help = "auto-validates phrases"
+
+    def handle(self, *args, **options):
+        """
+        Iterates over all the phrases in the database
+        and auto-standardizes wordforms where there is exactly one
+        matching analysis
+        """
+        phrases = Phrase.objects.all()
+        for phrase in tqdm(phrases):
```

Comment:
If there are a *lot* of phrases and takes a long time to even get started, `phrases.iterator()` goes through phrases in batches of 1000 instead of loading them all in memory up front.

Reply:
There are _a lot_ of phrases! I'll add this is now, but sadly we won't see how much faster it makes it for a long time...if ever.

Reply:
Hmm this seems to disrupt `tqdm` somehow, though. It doesn't show the full progress bar when I add the `.iterator()`

Reply:
Oops, sorry, [the way I do this is](https://github.com/UAlbertaALTLab/cree-intelligent-dictionary/blob/21018addd9e9b5921c27f26b9d7eb4a5a0b2105f/CreeDictionary/phrase_translate/management/commands/translatewordforms.py#L150-L151),

  for blah in tqdm(queryset.iterator(), total=queryset.count()):

This does the `count()` once at the start, which is faster than retrieving everything and jamming it all into memory, then page through in batches of 1000. If the database was actively being written to, the initial count could be different from how many items you actually iterate through, but if all that affects is the progress meter, no biggie.

Reply:
Perfect, that does it. I can't tell the difference locally since I have <1000 entries on my laptop, but I'm sure it'll make a difference when running this against the entire DB.